kore/src/Kore/Internal/Alias.hs

```diff
@@ -25,40 +25,50 @@ import Kore.Debug
 import Kore.Internal.ApplicationSorts
 import Kore.Sort
 import Kore.Syntax.Application
+import Kore.Syntax.Variable
+       ( Variable )
 import Kore.Unparser
 
-data Alias =
+data Alias patternType =
     Alias
         { aliasConstructor :: !Id
         , aliasParams      :: ![Sort]
         , aliasSorts       :: !ApplicationSorts
+        , aliasLeft        :: [Variable]
+        , aliasRight       :: patternType
         }
     deriving (GHC.Generic, Show)
 
-instance Eq Alias where
+instance Eq patternType => Eq (Alias patternType) where
     (==) a b =
             Function.on (==) aliasConstructor a b
         &&  Function.on (==) aliasParams a b
+        &&  Function.on (==) aliasSorts a b
```

Comment:
The comparisons with `(==)` and `compare` of `aliasSorts`, `aliasLeft`, and `aliasRight` should go away, too. The aliases are identically equal if `aliasConstructor` and `aliasParams` are equal.